// @universityofmaryland/web-styles-library/dist/element/text/decoration.js
import { color, font, spacing, media } from '../../token';
import { transition } from '../../animation';
import { elements } from '../../typography';
import { create } from '../../utilities';
const classNamePrefix = 'umd-text-decoration';
const watermarkChild = {
    position: 'absolute',
    top: '20px',
    left: '-2%',
    color: color.gray.lighter,
    fontWeight: 'bold',
    textTransform: 'uppercase',
    fontSize: `min(calc(${font.size['5xl']} + 13vw), 240px)`,
    lineHeight: '0',
    opacity: ' 0.6',
    pointerEvents: 'none',
    userSelect: 'none',
    zIndex: '-1',
    [`@media (${media.queries.large.max})`]: {
        display: 'none',
    },
};
export const watermark = create.jssObject({
    className: [
        `${classNamePrefix}-watermark`,
        'umd-watermark',
    ],
    position: 'relative',
    '> *': {
        ...watermarkChild,
        ...transition.slideRight,
    },
});
export const watermarkDark = create.jssObject({
    ...watermark,
    className: [
        `${classNamePrefix}-watermark-dark`,
        'umd-watermark-dark',
    ],
    '> *': {
        ...watermarkChild,
        ...transition.slideRight,
        opacity: '0.12',
        zIndex: 'inherit',
    },
});
export const ribbon = create.jssObject({
    className: [
        `${classNamePrefix}-eyebrow`,
        'umd-eyebrow-ribbon',
    ],
    ...elements.eyebrow,
    backgroundColor: color.gold,
    padding: `${spacing.min} ${spacing.md}`,
    display: `inline-block`,
    clipPath: `polygon(8% 0, 100% 0, 92% 100%, 0 100%)`,
});
