// @universityofmaryland/web-styles-library/dist/layout/background/quarter.js
import { color, media } from '../../token';
import { create } from '../../utilities';
import { padding } from './full';
const classNamePrefix = 'umd-layout-background-quarter';
const psuedo = {
    content: '""',
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    width: '100%',
    zIndex: -1,
    [`@media (${media.queries.tablet.max})`]: {
        display: 'none',
    },
    [`@media (${media.queries.desktop.min})`]: {
        width: `calc(75% + 80px)`,
    },
};
const quarter = {
    position: 'relative',
    ...padding,
    '&:before': {
        ...psuedo,
        backgroundColor: `${color.gray.lightest}`,
    },
};
export const light = create.jssObject({
    ...quarter,
    className: [
        `${classNamePrefix}-light`,
        'umd-layout-background-quater-color',
    ],
});
export const dark = create.jssObject({
    ...quarter,
    '&:before': {
        ...psuedo,
        backgroundColor: `${color.gray.darker}`,
    },
    className: [
        `${classNamePrefix}-dark`,
        'umd-layout-background-quater-color',
    ],
});
