// @universityofmaryland/web-styles-library/dist/layout/space/columns.js
import { color, media, spacing } from '../../token';
import { create } from '../../utilities';
const classNamePrefix = 'umd-layout-space-columns';
export const left = create.jssObject({
    [`@media (${media.queries.tablet.min})`]: {
        display: 'flex',
    },
    '& > *:first-child': {
        [`@media (${media.queries.tablet.max})`]: {
            display: 'none',
        },
        [`@media (${media.queries.tablet.min})`]: {
            marginRight: spacing['max'],
            width: '242px',
        },
    },
    '& > *:last-child': {
        [`@media (${media.queries.tablet.min})`]: {
            width: 'calc(100% - 242px)',
        },
    },
    className: [
        `${classNamePrefix}-left`,
        'umd-layout-interior-navigation',
    ],
});
export const right = create.jssObject({
    ...left,
    [`@media (${media.queries.tablet.min})`]: {
        display: 'flex',
    },
    '& > *:first-child': {
        [`@media (${media.queries.tablet.min})`]: {
            width: `calc(100% - 322px)`,
        },
    },
    '& > *:last-child': {
        paddingTop: spacing.md,
        borderTop: `1px solid ${color.black}`,
        [`@media (${media.queries.large.max})`]: {
            marginTop: spacing['3xl'],
        },
        [`@media (${media.queries.tablet.min})`]: {
            marginLeft: spacing['7xl'],
            width: '322px',
        },
    },
    className: [
        `${classNamePrefix}-right`,
        'umd-layout-interior-sidebar',
    ],
});
