// @universityofmaryland/web-styles-library/dist/layout/space/horizontal.js
import { media, spacing } from '../../token';
import { create } from '../../utilities';
const classNamePrefix = 'umd-layout-space-horizontal';
const lockBase = {
    display: 'block',
    marginLeft: 'auto',
    marginRight: 'auto',
    paddingLeft: spacing['md'],
    paddingRight: spacing['md'],
    maxWidth: `${spacing.maxWidth.normal}`,
    [`@media (${media.queries.tablet.min})`]: {
        paddingLeft: spacing['2xl'],
        paddingRight: spacing['2xl'],
    },
    [`@media (${media.queries.desktop.min})`]: {
        paddingLeft: spacing['4xl'],
        paddingRight: spacing['4xl'],
    },
    [`@media (${media.queries.highDef.min})`]: {
        paddingLeft: spacing.max,
        paddingRight: spacing.max,
    },
};
export const lockFull = create.jssObject({
    className: ['umd-lock-full'],
});
export const full = create.jssObject({
    ...lockBase,
    maxWidth: '100%',
    className: [
        `${classNamePrefix}-full`,
        'umd-lock-full',
    ],
});
export const lock = create.jssObject({
    className: ['umd-lock'],
});
export const max = create.jssObject({
    ...lockBase,
    maxWidth: `${spacing.maxWidth.max}`,
    className: [
        `${classNamePrefix}-larger`,
        'umd-lock',
    ],
});
export const lockLarge = create.jssObject({
    className: ['umd-lock-small'],
});
export const large = create.jssObject({
    ...lockBase,
    maxWidth: `${spacing.maxWidth.large}`,
    className: [
        `${classNamePrefix}-large`,
        'umd-lock-small',
    ],
});
export const lockNormal = create.jssObject({
    className: ['umd-lock-normal'],
});
export const normal = create.jssObject({
    ...lockBase,
    maxWidth: `${spacing.maxWidth.normal}`,
    className: [
        `${classNamePrefix}-normal`,
        'umd-lock-smaller',
    ],
});
export const lockSmall = create.jssObject({
    className: ['umd-lock-extra-small'],
});
export const small = create.jssObject({
    ...lockBase,
    maxWidth: `${spacing.maxWidth.small}`,
    className: [
        `${classNamePrefix}-small`,
        'umd-lock-extra-small',
    ],
});
export const lockSmallest = create.jssObject({
    className: ['umd-lock-smallest'],
});
export const smallest = create.jssObject({
    ...lockBase,
    maxWidth: `${spacing.maxWidth.smallest}`,
    className: [
        `${classNamePrefix}-smallest`,
        'umd-lock-smallest',
    ],
});
