// @universityofmaryland/web-styles-library/dist/layout/space/vertical.js
import { media, spacing } from '../../token';
import { create } from '../../utilities';
const classNamePrefix = 'umd-layout-space-vertical';
export const landing = create.jssObject({
    marginBottom: spacing['3xl'],
    [`@media (${media.queries.tablet.min})`]: {
        marginBottom: spacing['6xl'],
    },
    [`@media (${media.queries.desktop.min})`]: {
        marginBottom: '120px',
    },
    className: [
        `${classNamePrefix}-landing`,
        'umd-layout-vertical-landing',
    ],
});
export const landingChild = create.jssObject({
    marginBottom: spacing.lg,
    [`@media (${media.queries.tablet.min})`]: {
        marginBottom: spacing['xl'],
    },
    [`@media (${media.queries.desktop.min})`]: {
        marginBottom: spacing['2xl'],
    },
    className: [
        `${classNamePrefix}-landing-child`,
        'umd-layout-vertical-landing-child',
    ],
});
export const interior = create.jssObject({
    marginBottom: spacing['3xl'],
    [`@media (${media.queries.desktop.min})`]: {
        marginBottom: spacing['6xl'],
    },
    className: [
        `${classNamePrefix}-interior`,
        'umd-layout-vertical-interior',
    ],
});
export const interiorChild = create.jssObject({
    marginBottom: spacing.lg,
    className: [
        `${classNamePrefix}-interior-child`,
        'umd-layout-vertical-interior-child',
    ],
});
export const headlineLarge = create.jssObject({
    marginBottom: spacing.sm,
    [`@media (${media.queries.desktop.min})`]: {
        marginBottom: spacing.md,
    },
    className: [
        `${classNamePrefix}-headline-large`,
        'umd-layout-headline-large',
    ],
});
export const headlineMedium = create.jssObject({
    marginBottom: spacing.sm,
    className: [
        `${classNamePrefix}-headline-medium`,
        'umd-layout-headline-medium',
    ],
});
