// @universityofmaryland/web-styles-library/dist/web-components.js
import { color, media, spacing } from './token';
const createElementStyles = (elementName, additionalStyles = {}) => ({
    [`${elementName}:not(:defined)`]: {
        display: 'none',
        ...additionalStyles.notDefined,
    },
    [`${elementName}:defined`]: {
        display: 'block',
        containerType: 'inline-size',
        ...additionalStyles.defined,
    },
    ...additionalStyles.custom,
});
const baseElements = [
    'alert-page',
    'alert-site',
    'banner-promo',
    'carousel-cards',
    'carousel-image',
    'carousel-multiple-image',
    'carousel-people',
    'carousel-thumbnail',
    'carousel',
    'feed-news',
    'hero-brand-video',
    'hero-expand',
    'hero-logo',
    'hero-minimal',
    'hero',
    'logo',
    'media-inline',
    'pathway-highlight',
    'pathway',
    'person-bio',
    'quote',
    'slider-events-feed',
    'slider-events',
    'tabs',
].reduce((acc, name) => ({
    ...acc,
    ...createElementStyles(`umd-element-${name}`),
}), {});
const elements = {
    ...baseElements,
    ...createElementStyles('umd-element-accordion-item', {
        custom: {
            'umd-element-accordion-item + umd-element-accordion-item': {
                marginTop: spacing.min,
            },
        },
    }),
    ...createElementStyles('umd-element-brand-logo-animation', {
        defined: {
            position: 'absolute',
            right: '0',
            top: '-3vw',
            height: '25vw',
            zIndex: '9',
            overflow: 'clip',
            containerType: 'normal',
            [`@media (${media.queries.highDef.min})`]: {
                top: '-11vw',
                height: '50vw',
            },
        },
    }),
    ...['article', 'card', 'card-overlay', 'event', 'person'].reduce((acc, type) => ({
        ...acc,
        ...createElementStyles(`umd-element-${type}`, {
            custom: {
                [`umd-element-${type}[display="list"] + umd-element-${type}[display="list"]`]: {
                    marginTop: spacing.md,
                },
            },
        }),
    }), {}),
    ...['person'].reduce((acc, type) => ({
        ...acc,
        ...createElementStyles(`umd-element-${type}`, {
            custom: {
                [`umd-element-${type}[display="tabular"] + umd-element-${type}[display="tabular"]`]: {
                    marginTop: spacing.md,
                },
            },
        }),
    }), {}),
};
const breadcrumb = createElementStyles('umd-element-breadcrumb', {
    defined: {
        display: 'block',
        marginTop: `${spacing.sm}`,
    },
});
const action = createElementStyles('umd-element-call-to-action', {
    defined: {
        containerType: 'normal',
    },
});
const footer = createElementStyles('umd-element-footer', {
    notDefined: {
        backgroundColor: color.black,
        height: `calc(${spacing.md} * 20)`,
        '& > *': { display: 'none' },
    },
});
const navigation = {
    ...['nav-item', 'navigation-header'].reduce((acc, name) => ({
        ...acc,
        ...createElementStyles(`umd-element-${name}`, {
            defined: {
                containerType: 'normal',
            },
        }),
    }), {}),
    ...createElementStyles('umd-element-nav-drawer', {
        notDefined: {
            backgroundColor: color.white,
            height: '44px',
            width: '44px',
            '& > *': { display: 'none' },
        },
    }),
};
const navigationSticky = {
    ...['navigation-sticky'].reduce((acc, name) => ({
        ...acc,
        ...createElementStyles(`umd-element-${name}`, {
            custom: {
                [`umd-element-${name}`]: {
                    position: 'relative',
                    zIndex: '9999',
                },
            },
        }),
    }), {}),
};
const scrollTopTag = 'umd-element-scroll-top';
const scrollTop = createElementStyles(scrollTopTag, {
    defined: {
        height: '40px',
        width: '40px',
    },
    custom: {
        [`* + ${scrollTopTag}`]: { marginTop: spacing.md },
        [`${scrollTopTag}[fixed], ${scrollTopTag}[data-layout-fixed="true"]`]: {
            position: 'fixed',
            right: '8px',
            bottom: '10vh',
            zIndex: '9999',
            [`@media (${media.queries.tablet.min})`]: {
                height: '52px',
                width: '52px',
                right: '40px',
            },
        },
    },
});
const sectionIntro = {
    ...['section-intro-wide', 'section-intro'].reduce((acc, name) => ({
        ...acc,
        ...createElementStyles(`umd-element-${name}`, {
            custom: {
                [`umd-element-section-intro-wide + *, umd-element-section-intro + *`]: {
                    marginTop: spacing.lg,
                    [`@media (${media.queries.tablet.min})`]: {
                        marginTop: `${spacing['2xl']}`,
                    },
                },
            },
        }),
    }), {}),
};
const socialSharingTag = 'umd-element-social-sharing';
const socialSharing = createElementStyles(socialSharingTag, {
    custom: {
        [`* + ${socialSharingTag}`]: { marginTop: spacing.md },
        [`${socialSharingTag}[fixed], ${socialSharingTag}[data-layout-fixed="true"]`]: {
            [`@media (${media.queries.tablet.min})`]: {
                position: 'fixed',
                left: '0',
                top: '30vh',
                zIndex: '9999',
            },
            [`@media (${media.queries.desktop.min})`]: {
                left: '40px',
            },
        },
    },
});
export default {
    ...elements,
    ...breadcrumb,
    ...action,
    ...footer,
    ...navigation,
    ...navigationSticky,
    ...scrollTop,
    ...sectionIntro,
    ...socialSharing,
};
