// @universityofmaryland/web-styles-library/dist/layout/grid/base.js
import { media, spacing } from '../../token';
import { create } from '../../utilities';
const classNamePrefix = 'umd-layout-grid-columns';
const baseSmallScreen = {
    display: 'grid',
    [`@media (${media.queries.medium.max})`]: {
        gridGap: spacing.lg,
    },
};
export const base = {
    two: {
        ...baseSmallScreen,
        [`@media (${media.queries.large.min})`]: {
            gridTemplateColumns: 'repeat(2, 1fr)',
        },
    },
    three: {
        ...baseSmallScreen,
        [`@media (${media.queries.desktop.min})`]: {
            gridTemplateColumns: 'repeat(3, 1fr)',
        },
    },
    four: {
        ...baseSmallScreen,
        [`@media (${media.queries.large.min})`]: {
            gridTemplateColumns: 'repeat(2, 1fr)',
        },
        [`@media (${media.queries.desktop.min})`]: {
            gridTemplateColumns: 'repeat(4, 1fr)',
        },
    },
};
export const columnsTwo = create.jssObject({
    ...base.two,
    className: [
        `${classNamePrefix}-two`,
        'umd-grid',
    ],
});
export const columnsThree = create.jssObject({
    ...base.three,
    className: [
        `${classNamePrefix}-three`,
        'umd-grid-three',
    ],
});
export const columnsFour = create.jssObject({
    ...base.four,
    className: [
        `${classNamePrefix}-four`,
        'umd-grid-four',
    ],
});
export const stacked = create.jssObject({
    display: 'flex',
    flexDirection: 'column',
    className: `${classNamePrefix}-stacked`,
});
