// @universityofmaryland/web-styles-library/dist/layout/grid/border.js
import { color, media } from '../../token';
import { create } from '../../utilities';
import { base } from './base';
const classNamePrefix = 'umd-layout-grid-border';
const boarderBase = {
    border: `1px solid ${color.gray.light}`,
    borderBottom: `none`,
    [`@media (${media.queries.medium.max})`]: {
        gridGap: 0,
    },
};
export const columnsTwo = create.jssObject({
    ...base.two,
    ...boarderBase,
    className: [
        `${classNamePrefix}-two`,
        'umd-grid-border',
    ],
    '& > *': {
        borderBottom: `1px solid ${color.gray.light}`,
        borderRight: `1px solid ${color.gray.light}`,
    },
    '&:not(:has(> :last-child:nth-child(2)))': {
        borderTop: `none`,
        borderRight: `none`,
        '& > *:first-child': {
            borderTop: `1px solid ${color.gray.light}`,
        },
        '& > *:nth-child(2)': {
            borderTop: `1px solid ${color.gray.light}`,
        },
    },
});
export const columnsThree = create.jssObject({
    ...base.three,
    ...boarderBase,
    className: [
        `${classNamePrefix}-three`,
        'umd-grid-three-border',
    ],
    '& > *': {
        borderBottom: `1px solid ${color.gray.light}`,
        borderRight: `1px solid ${color.gray.light}`,
    },
    '&:not(:has(> :last-child:nth-child(3)))': {
        borderTop: `none`,
        borderRight: `none`,
        '& > *:first-child': {
            borderTop: `1px solid ${color.gray.light}`,
        },
        '& > *:nth-child(2)': {
            borderTop: `1px solid ${color.gray.light}`,
        },
        '& > *:nth-child(3)': {
            borderTop: `1px solid ${color.gray.light}`,
        },
    },
});
export const columnsFour = create.jssObject({
    ...base.four,
    ...boarderBase,
    className: [
        `${classNamePrefix}-four`,
        'umd-grid-four-border',
    ],
    '& > *': {
        borderBottom: `1px solid ${color.gray.light}`,
        borderRight: `1px solid ${color.gray.light}`,
    },
    '&:not(:has(> :last-child:nth-child(4)))': {
        borderTop: `none`,
        borderRight: `none`,
        '& > *:first-child': {
            borderTop: `1px solid ${color.gray.light}`,
        },
        '& > *:nth-child(2)': {
            borderTop: `1px solid ${color.gray.light}`,
        },
        '& > *:nth-child(3)': {
            borderTop: `1px solid ${color.gray.light}`,
        },
        '& > *:nth-child(4)': {
            borderTop: `1px solid ${color.gray.light}`,
        },
    },
});
