// @universityofmaryland/web-styles-library/dist/layout/grid/inline.js
import { media, spacing } from '../../token';
import { create } from '../../utilities';
const classNamePrefix = 'umd-layout-grid-inline';
const columns = {
    alignItems: 'flex-start',
    display: 'flex',
    flexDirection: 'column',
    gap: spacing.sm,
    justifyContent: 'flex-start',
};
export const row = {
    className: `${classNamePrefix}-rows`,
    display: 'flex',
    gap: spacing.min,
    alignItems: 'center',
};
export const tabletRows = create.jssObject({
    ...columns,
    [`@media (${media.queries.large.min})`]: {
        flexDirection: 'row',
        flexWrap: 'wrap',
        alignItems: 'center',
    },
    className: [
        `${classNamePrefix}-tablet-rows`,
        'umd-grid-row-mobile-tablet',
    ],
});
export const stretch = create.jssObject({
    display: 'flex',
    flexWrap: 'wrap',
    gap: spacing.sm,
    justifyContent: 'space-between',
    position: 'relative',
    '& > *:first-child': {
        maxWidth: '100%',
        flex: '1 0 auto',
    },
    '& > *:first-child + *:last-child:not(:first-child)': {
        margin: 0,
        position: 'relative',
        zIndex: 999,
    },
    className: [
        `${classNamePrefix}-stretch`,
        'umd-forms-layout-headline-with-action',
    ],
});
