// @universityofmaryland/web-styles-library/dist/element/composite/person/bio.js
import { media, spacing } from '../../../token';
import { create } from '../../../utilities';
const classNamePrefix = 'umd-element-composite-person-bio';
const createContainerQuery = (breakpoint, comparison = 'max-width', styles = {}) => {
    return {
        [`@container (${comparison}: ${breakpoint}px)`]: styles,
    };
};
const bioLayout = {
    display: 'grid',
    gridGap: `${spacing.md}`,
    ...createContainerQuery(media.breakpointValues.large.min, 'min-width', {
        gridTemplateColumns: `repeat(8, 1fr)`,
        gridGap: `${spacing.lg}`,
        alignItems: `center`,
    }),
    [`& img`]: {
        ...createContainerQuery(media.breakpointValues.large.min, 'min-width', {
            width: `100%`,
            height: `auto !important`,
        }),
    },
    [`& > *`]: {
        ...createContainerQuery(media.breakpointValues.large.min, 'min-width', {
            gridColumn: `span 5`,
        }),
    },
    [`&:has(> :nth-child(2)) > *:first-child `]: {
        ...createContainerQuery(media.breakpointValues.large.min, 'min-width', {
            gridColumn: `span 3`,
            alignSelf: `flex-start`,
        }),
    },
};
export const small = create.jssObject({
    className: `${classNamePrefix}-small`,
    ...bioLayout,
});
export const smallDark = create.jssObject({
    className: `${classNamePrefix}-small-dark`,
    ...bioLayout,
});
